### PR TITLE
Feature/5582 Event Check-in UI Scan QR Code/6181 UI Test

### PR DESCRIPTION
**Pull request by tcfos**
## Description
UITest for:
- scan a QR code
- check in
- check out

## Link to Jira
Task https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6181

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewController.xib [hunk 1]
```diff
@@ -113,7 +113,7 @@
                                                     <rect key="frame" x="16" y="30" width="350" height="161.5"/>
                                                     <subviews>
                                                         <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Vereinsaktivität " textAlignment="natural" lineBreakMode="tailTruncation" numberOfLines="0" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="Jzq-Fo-8cl" customClass="ENALabel" customModule="ENA" customModuleProvider="target">
-                                                            <rect key="frame" x="0.0" y="0.0" width="350" height="20.5"/>
+                                                            <rect key="frame" x="0.0" y="0.0" width="350" height="80"/>
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewController.xib:116]: are these changes intended ? was there a problem with the constrains or what?

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift [hunk 2]
```diff
@@ -18,6 +18,36 @@ class ENAUITests_09_CheckIns: XCTestCase {
 		
 	}
 	
+	func test_WHEN_scan_QRCode_THEN_checkin_and_checkout() {
+		// GIVEN
+		app.launchArguments.append(contentsOf: ["-checkinInfoScreenShown", "YES"])
+		app.launch()
+				XCTAssertTrue(app.buttons[AccessibilityIdentifiers.Tabbar.checkin].waitForExistence(timeout: .short))
```
Comment (naveeddotio) [inline src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift:25]: wrong indentation or github?

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift [hunk 3]
```diff
@@ -18,6 +18,36 @@ class ENAUITests_09_CheckIns: XCTestCase {
...
+		XCTAssertTrue(app.staticTexts["Walldorf"].exists)
+		XCTAssertTrue(app.staticTexts[AppStrings.TraceLocations.unspecified.title].exists)
+		
+		// check in
+		XCTAssertTrue(app.buttons[AccessibilityIdentifiers.TraceLocation.Details.checkInButton].waitForExistence(timeout: .short))
+		app.buttons[AccessibilityIdentifiers.TraceLocation.Details.checkInButton].tap()
+		
+		// check out and clean up
+		myCheckins_checkout()
+
```
Comment (naveeddotio) [inline src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift:48]: extra space can be removed

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift [hunk 4]
```diff
@@ -61,5 +91,65 @@ class ENAUITests_09_CheckIns: XCTestCase {
...
+		let query = app.cells.buttons
+		let n = query.count
+		XCTAssertTrue(n > 1)
+		var numberOfCheckouts = 0
+		for i in 0...(n - 1) {
+			if query.element(boundBy: i).identifier == AccessibilityIdentifiers.TraceLocation.Configuration.eventTableViewCellButton {
+				numberOfCheckouts = numberOfCheckouts.inc()
+			}
+		}
+		XCTAssertTrue( numberOfCheckouts == 1 ) // assumption: one cell has a checkout button
```
Comment (naveeddotio) [inline src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift:111]: extra spaces in an argument

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift [hunk 5]
```diff
@@ -61,5 +91,65 @@ class ENAUITests_09_CheckIns: XCTestCase {
...
+		
+		// tap checkout button
+		XCTAssertTrue(query.element(boundBy: 1).identifier == AccessibilityIdentifiers.TraceLocation.Configuration.eventTableViewCellButton)
+		query.element(boundBy: 1).tap()
+		
+		// tap the event, verify the detail screen
+		XCTAssertTrue( initialNumberOfCells == app.cells.count ) // assumption: number of cells has not changed
```
Comment (naveeddotio) [inline src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift:118]: extra spaces in an argument

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewController.xib (hunk 1 above)
Comment (tcfos) [reply src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewController.xib]: Happened accidentally. Fixed with 8269bef

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift (hunk 2 above)
Comment (tcfos) [reply src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift]: Code formatting done.

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift (hunk 3 above)
Comment (tcfos) [reply src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift]: Code formatting done.

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift (hunk 4 above)
Comment (tcfos) [reply src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift]: Code formatting done.

File: src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift (hunk 5 above)
Comment (tcfos) [reply src/xcode/ENA/ENAUITests/CheckIns/ENAUITests_09_CheckIns.swift]: Code formatting done.

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeScanner/CheckinQRCodeScannerViewModel.swift [hunk 6]
```diff
@@ -15,6 +15,21 @@ class CheckinQRCodeScannerViewModel: NSObject, AVCaptureMetadataOutputObjectsDel
...
+				description: "Supermarkt",
+				address: "Walldorf",
+				startDate: nil,
+				endDate: nil,
+				defaultCheckInLengthInMinutes: nil,
+				cryptographicSeed: Data(),
+				cnPublicKey: Data()
+			)
+			onSuccess(traceLocation)
+		}
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeScanner/CheckinQRCodeScannerViewModel.swift:32]: that should be surrounded by #if DEBUG

**Review by pascal-brause — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeScanner/CheckinQRCodeScannerViewModel.swift (hunk 6 above)
Comment (tcfos) [reply src/xcode/ENA/ENA/Source/Scenes/Checkin/QRCodeScanner/CheckinQRCodeScannerViewModel.swift:33]: Oh, thx. Fixed with 9c67014

**Review by pascal-brause — APPROVED** [review]

**Review by marcussc — APPROVED** [review]

**Review by naveeddotio — APPROVED** [review]
